src/config.js

```diff
@@ -0,0 +1,203 @@
+      }
+      this.startingAddressIndex = startingAddressIndex;
+    }
+  }
+
+  /**
+   * @description instantiate a new wallet config object from a JSON
+   * @param {string} jsonString - json string to convert.
+   * @returns {MultisigWalletConfig} returns a new instance of a MultisigWalletConfig
+   */
+  static fromJSON(jsonString) {
+    if (typeof jsonString !== 'string') throw new TypeError('Must pass a valid JSON string');
+    let options;
+
+    try {
+      options = JSON.parse(jsonString);
+    } catch (error) {
+      throw new Error("Unable to parse JSON.");
+    }
+    
+    // TODO: how do we handle caravan configs where the extended public keys
+    // don't include xfp information? We can get a parent fingerprint 
+    // from the xpub and use that. It wouldn't be valid as an xfp but could be used as stand in? 
+    return new this(options);
+  }
+
+  /**
+   * @description static method wrapper around getting a new
+   * config based on a JSON since Caravan is the assumed format of accepted JSON
+   * @param {string} jsonString - valid config json string from Caravan export
+   * @returns {MultisigWalletConfig} new instance of a MultisigWalletConfig
+   */
+  static fromCaravanConfig(jsonString) {
+    return MultisigWalletConfig.fromJSON(jsonString);
+  }
+
+  /**
+   * @description Validate the public keys on the config instances. 
+   * Not all configs require xfp in the xpub objects, so an optional param is
+   * accepted to dictate whether or not this value should be validated
+   * @param {boolean} requiresXFP - does the object require an xfp, if so it will also be validated
+   * @returns {boolean|Error} true if valid otherwise throws an error
+   */
+  validateExtendedPublicKeys(requiresXFP=false) {
+    return this.extendedPublicKeys.every(({ xpub, bip32Path, xfp }) => {
+      assert(xpub, 'xpub value required');
+
+      let xpubError = validateExtendedPublicKeyForNetwork(xpub, this.network);
+      assert(!xpubError.length, `Error in Xpub ${xpub}: ${xpubError}`)
+
+      if (bip32Path && bip32Path !== 'Unknown') {
+        const pathError = validateBIP32Path(bip32Path);
+        assert(!pathError.length, `Xpub Path Error: ${bip32Path} - ${pathError}`)
+      }
+
+      if (requiresXFP) {
+        assert(xfp && xfp !== "Unknown", "ExtendedPublicKeys missing at least one xfp.");
+        validateRootFingerprint(xfp)
+      }
+
+      return true
+    })
+  }
+
+  /**
+   * @description A simple wrapper for returning a json config
+   * since the params of the class are valid for caravan.
+   * @returns {MultisigWalletConfig} a new MultisigWalletConfig instance
+   */
+  toCaravanConfig() {
+    return this.toJSON();
+  }
+
+  /**
+   * @returns {string} returns a JSON string with config values
+   */
+  toJSON() {
+    return JSON.stringify({
+      name: this.name,
+      addressType: this.addressType,
+      network: this.network,
+      quorum: this.quorum,
```

Comment:
Note: This needs to be fixed.`quorum` is no longer in the MultisigWalletConfig properties, but is in the Caravan config. Needs to be inferred from other data for `CaravanConfig` and here should be replaced with `requiredSigners`